ccips/ccip-008/ccip-008-citycoins-sip-010-token-v2.md

```diff
@@ -0,0 +1,113 @@
+
+- the number of Stacks blocks before a halving occurs
+- the Stacks block thresholds set for each halving
+- the activation status of the token contract
+- the core contract states
+
+### Token Activation
+
+Once a CityCoin is deployed and activated, the emission schedule begins starting from the activation block height of the CityCoin core contract.
+
+More information about Activation can be found in [CCIP-002](../ccip-002/ccip-002-citycoins-activation.md).
+
+CityCoins have no ICO, no pre-sale, and no pre-mine. CityCoins are only mined into existence.
+
+### Emissions Schedule
+
+Miners receive coinbase rewards for mining a CityCoin outlined in the table below. The "halvings" are based on a doubling epoch model with an initial bonus period, in which the length of each epoch is twice as long as the last following epoch 1.
+
+Following the token activation, the emissions schedule continues as follows:
+
+| Epoch | Epoch Length | Epoch End Block | Block Reward | Total Supply   |
+| ----- | ------------ | --------------- | ------------ | -------------- |
+| 0     | 10,000       | 10,000          | 250,000      | 2,500,000,000  |
+| 1     | 25,000       | 35,000          | 100,000      | 5,000,000,000  |
+| 2     | 50,000       | 85,000          | 50,000       | 7,500,000,000  |
+| 3     | 100,000      | 185,000         | 25,000       | 10,000,000,000 |
+| 4     | 200,000      | 385,000         | 12,500       | 12,500,000,000 |
+| 5     | 400,000      | 785,000         | 6,250        | 15,000,000,000 |
+| 6     | 800,000      | 1,585,000       | 3,125        | 17,500,000,000 |
+
+Note: the values above are denoted in CityCoins, and the values in the contract and APIs the value will represent the reward/supply above multiplied by `1,000,000` to account for the 6 decimal places as micro-CityCoins.
+
+Included with this CCIP is a [supplemental spreadsheet](./ccip-008-0001-doubling-epoch-emissions-schedule.ods) that shows the calculations for and graphs the differences between the old and new emissions schedule.
+
+## Backwards Compatibility
+
+This CCIP replaces the token standard in CCIP-005 and is not backwards compatible, however any existing CityCoin deployed based on CCIP-005 can be converted to the equivalent CityCoin deployed based on procedures set forth in this CCIP.
+
+The implementation needs to occur before Stacks block 59,497 in order for the total supply to be in sync with future CityCoins.
+
+### Token Conversion
+
+The CCIP-005 token contracts for MIA[^4] and NYC[^5] are currently deployed to the Stacks mainnet.
+
+The CCIP-008 token contracts will be deployed to the Stacks mainnet and replace the functionality of the CCIP-005 token contracts.
+
+In order to make the upgrade as seamless as possible, contract functions will be provided that:
+
+- allow direct conversion by burning the CCIP-005 asset and minting the CCIP-008 asset
+- allow conversion through mining and stacking within the newly deployed core contract
```

Comment:
IMHO this should not be even considered. Every little addition to mining/stacking code will increase execution costs.
Conversion should be handled by separate contract and executed manually be each user.

Reply:
Totally understand that! Originally I was thinking of trying to hook it up *in the same function*, but realized it would be hard to differentiate the assets at that point.

The new thought here was to have an extra function called something like `mine-and-convert` or `stack-and-convert` that accepts the old token, converts it, then calls the relevant private function in the core. The UI could detect which to call based on what asset the user has. 

This adds slightly to the overall read length for the contract, but I'm hoping it won't be much code to achieve the result.